libraries/Adapters/Microsoft.Bot.Builder.Adapters.WeChat/WeChatHttpAdapter.cs

```diff
@@ -0,0 +1,418 @@
+                };
+            }
+
+            if (botStateSet != null)
+            {
+                Use(new AutoSaveStateMiddleware(botStateSet));
+            }
+        }
+
+        /// <summary>
+        /// Process the request from WeChat.
+        /// </summary>
+        /// <param name="wechatRequest">Request message entity from wechat.</param>
+        /// <param name="callback"> Bot callback handler.</param>
+        /// <param name="passiveResponse">Marked the message whether it needs passive reply or not. </param>
+        /// <param name="cancellationToken">Cancellation Token of this Task.</param>
+        /// <returns>Response message entity.</returns>
+        public async Task<object> ProcessWeChatRequest(IRequestMessageBase wechatRequest, BotCallbackHandler callback, bool passiveResponse, CancellationToken cancellationToken)
+        {
+            var activity = await _wechatMessageMapper.ToConnectorMessage(wechatRequest).ConfigureAwait(false);
+            BotAssert.ActivityNotNull(activity);
+            using (var context = new TurnContext(this, activity as Activity))
+            {
+                try
+                {
+                    var responses = new Dictionary<string, List<Activity>>();
+                    context.TurnState.Add(TurnResponseKey, responses);
+                    await RunPipelineAsync(context, callback, cancellationToken).ConfigureAwait(false);
+                    var key = $"{activity.Conversation.Id}:{activity.Id}";
+                    try
+                    {
+                        var activities = responses.ContainsKey(key) ? responses[key] : new List<Activity>();
+                        var response = await ProcessBotResponse(activities, wechatRequest.FromUserName, passiveResponse).ConfigureAwait(false);
+                        return response;
+                    }
+                    catch (Exception e)
+                    {
+                        // TODO: exception handling when send message to wechat api failed.
+                        _logger.LogError(e, "Failed to process bot response");
+                        throw;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // exception handing when bot throw an exception.
+                    await OnTurnError(context, ex).ConfigureAwait(false);
+                    return null;
+                }
+            }
+        }
+
+        // Does not support by wechat.
+        public override Task DeleteActivityAsync(ITurnContext turnContext, ConversationReference reference, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        // Does not support by wechat.
+        public override Task<ResourceResponse> UpdateActivityAsync(ITurnContext turnContext, Activity activity, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
```

Comment:
Use NotSupportedException (instead of NotImplemented). 

The code should look like:

```
return Task.FromException<ResourceResponse>(new NotSupportedException("WeChat does not support updating activities."));
```